### ISSUE TITLE
Use a different storage strategy for .pyireport files

### ISSUE THREAD

**Issue by inducer** [joerick/pyinstrument#99]
* Using predictable patterns (such as `/tmp/pyinstrument`) is insecure: Someone can just drop a symlink there and make you overwrite random stuff.
* Using one directory (`/tmp/pyinstrument`) for all users will break on multi-user systems

**Comment by joerick** [comment]
Yes, good points. https://github.com/ActiveState/appdirs/blob/master/appdirs.py might work here, instead.
